Bump Java and dependency versions

Krymonota opened the pull request:
I'm currently working on adding [Velocity](https://www.velocitypowered.com/) support to this project. (I'll create another pull request for it, though.)

However, the build fails because the source and target versions of the Bukkit module are set to Java 6:
```
[ERROR] goal org.apache.maven.plugins:maven-compiler-plugin:3.1:compile (default-compile) on project blccpsapibukkit could not be executed:  Compilation error: Compilation error:  
ERROR] Source option 6 is no longer supported.  Use 7 or higher. 
ERROR] Target option 6 is no longer supported.  Use 7 or higher.
```

Updating the module to Java 7 fixed the issue. I took this opportunity to update the Maven Shade Plugin and Gson version as well. I tested both the Bukkit and the BungeeCord on my instances and everything still works.

Krymonota committed:
Upgrade Bukkit Maven source and target version from Java 6 to Java 7

Krymonota committed:
Upgrade Bukkit Maven Shade Plugin from 3.1.0 to 3.2.1

Krymonota committed:
Upgrade Bukkit Gson dependency from 2.6.2 to 2.8.6

Rigner commented on blccpsapibukkit/pom.xml:
```diff
@@ -35,14 +35,14 @@
-                    <source>1.6</source>
-                    <target>1.6</target>
+                    <source>1.7</source>
+                    <target>1.7</target>
```
I just tested it myself on a clean install, I can build it fine with Java 6.

This was kept in order to make sure this plugin is compatible with most setups, as some servers may still be using Java 6.

Can you provide us the complete Maven output ?

Krymonota replied:
The `maven-compiler-plugin` uses `javac` to compile the project. There's an OpenJDK policy outlined in [JEP 182: Policy for Retiring javac -source and -target Options](https://openjdk.java.net/jeps/182) for retiring old `javac` `-source` and `-target` options. Support for a source or target of `1.5` was removed in JDK 9 and to comply with this policy, [support for a source or target of `1.6` was already removed in JDK 12](https://bugs.java.com/bugdatabase/view_bug.do?bug_id=JDK-8028563). 

I use JDK 13 and JDK 14 will become generally available in the first quarter of 2020. 

[According to MCStats.org only 0.01% (10 of over 100,000 tracked servers) still use Java 6. Even Java 7 is barely used.](http://mcstats.org/global/)
![grafik](https://user-images.githubusercontent.com/12609161/69224498-f9062080-0b7c-11ea-8b75-7bd5d2d403a7.png)

[bStats.org doesn't even track any servers that are still running on Java 6 or 7.](https://bstats.org/global/bukkit)

And by the way, the `source` and `target` options of the BungeeCord plugin are also set to `1.7`.

For these reasons I'd strongly suggest to update to Java 7 so that third party developers (like me) can also contribute to this project.

```
[INFO] Scanning for projects...
[WARNING] 
[WARNING] Some problems were encountered while building the effective model for net.badlion:blccpsapibungee:jar:1.2.1
[WARNING] 'build.plugins.plugin.version' for org.apache.maven.plugins:maven-compiler-plugin is missing. @ line 34, column 21
[WARNING] 
[WARNING] Some problems were encountered while building the effective model for net.badlion:blccpsapibukkit:jar:1.2.1
[WARNING] 'build.plugins.plugin.version' for org.apache.maven.plugins:maven-compiler-plugin is missing. @ line 34, column 21
[WARNING] 
[WARNING] It is highly recommended to fix these problems because they threaten the stability of your build.
[WARNING] 
[WARNING] For this reason, future Maven versions might no longer support building such malformed projects.
[WARNING] 
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Build Order:
[INFO] 
[INFO] badlionclientclickspersecondapi                                    [pom]
[INFO] blccpsapibungee                                                    [jar]
[INFO] blccpsapibukkit                                                    [jar]
[INFO] 
[INFO] ------------< net.badlion:badlionclientclickspersecondapi >-------------
[INFO] Building badlionclientclickspersecondapi 1.2.1                     [1/3]
[INFO] --------------------------------[ pom ]---------------------------------
[INFO] 
[INFO] --- maven-clean-plugin:2.5:clean (default-clean) @ badlionclientclickspersecondapi ---
[INFO] 
[INFO] --------------------< net.badlion:blccpsapibungee >---------------------
[INFO] Building blccpsapibungee 1.2.1                                     [2/3]
[INFO] --------------------------------[ jar ]---------------------------------
[INFO] 
[INFO] --- maven-clean-plugin:2.5:clean (default-clean) @ blccpsapibungee ---
[INFO] Deleting C:\Users\nikla\eclipse\workspaces\soupgg\badlionclientcpsapi\blccpsapibungee\target
[INFO] 
[INFO] --- maven-resources-plugin:2.6:resources (default-resources) @ blccpsapibungee ---
[INFO] Using 'UTF-8' encoding to copy filtered resources.
[INFO] Copying 1 resource to .
[INFO] 
[INFO] --- maven-compiler-plugin:3.1:compile (default-compile) @ blccpsapibungee ---
[INFO] Changes detected - recompiling the module!
[INFO] Compiling 3 source files to C:\Users\nikla\eclipse\workspaces\soupgg\badlionclientcpsapi\blccpsapibungee\target\classes
[INFO] 
[INFO] --- maven-resources-plugin:2.6:testResources (default-testResources) @ blccpsapibungee ---
[INFO] Using 'UTF-8' encoding to copy filtered resources.
[INFO] skip non existing resourceDirectory C:\Users\nikla\eclipse\workspaces\soupgg\badlionclientcpsapi\blccpsapibungee\src\test\resources
[INFO] 
[INFO] --- maven-compiler-plugin:3.1:testCompile (default-testCompile) @ blccpsapibungee ---
[INFO] No sources to compile
[INFO] 
[INFO] --- maven-surefire-plugin:2.12.4:test (default-test) @ blccpsapibungee ---
[INFO] No tests to run.
[INFO] 
[INFO] --- maven-jar-plugin:2.4:jar (default-jar) @ blccpsapibungee ---
[INFO] Building jar: C:\Users\nikla\eclipse\workspaces\soupgg\badlionclientcpsapi\blccpsapibungee\target\badlionclientcpsapibungee.jar
[INFO] 
[INFO] --------------------< net.badlion:blccpsapibukkit >---------------------
[INFO] Building blccpsapibukkit 1.2.1                                     [3/3]
[INFO] --------------------------------[ jar ]---------------------------------
[INFO] 
[INFO] --- maven-clean-plugin:2.5:clean (default-clean) @ blccpsapibukkit ---
[INFO] Deleting C:\Users\nikla\eclipse\workspaces\soupgg\badlionclientcpsapi\blccpsapibukkit\target
[INFO] 
[INFO] --- maven-resources-plugin:2.6:resources (default-resources) @ blccpsapibukkit ---
[INFO] Using 'UTF-8' encoding to copy filtered resources.
[INFO] Copying 1 resource to .
[INFO] 
[INFO] --- maven-compiler-plugin:3.1:compile (default-compile) @ blccpsapibukkit ---
[INFO] Changes detected - recompiling the module!
[INFO] Compiling 3 source files to C:\Users\nikla\eclipse\workspaces\soupgg\badlionclientcpsapi\blccpsapibukkit\target\classes
[INFO] -------------------------------------------------------------
[ERROR] COMPILATION ERROR : 
[INFO] -------------------------------------------------------------
[ERROR] Source option 6 is no longer supported. Use 7 or later.
[ERROR] Target option 6 is no longer supported. Use 7 or later.
[INFO] 2 errors 
[INFO] -------------------------------------------------------------
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary for badlionclientclickspersecondapi 1.2.1:
[INFO] 
[INFO] badlionclientclickspersecondapi .................... SUCCESS [  0.121 s]
[INFO] blccpsapibungee .................................... SUCCESS [  1.490 s]
[INFO] blccpsapibukkit .................................... FAILURE [  0.087 s]
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  1.789 s
[INFO] Finished at: 2019-11-20T10:21:19+01:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.1:compile (default-compile) on project blccpsapibukkit: Compilation failure: Compilation failure: 
[ERROR] Source option 6 is no longer supported. Use 7 or later.
[ERROR] Target option 6 is no longer supported. Use 7 or later.
[ERROR] -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoFailureException
[ERROR] 
[ERROR] After correcting the problems, you can resume the build with the command
[ERROR]   mvn <goals> -rf :blccpsapibukkit
```

Krymonota replied:
I fixed the warnings from the beginning of the build log in a new commit (see below).

Rigner commented on blccpsapibukkit/pom.xml:
```diff
@@ -65,7 +65,7 @@
             <groupId>com.google.code.gson</groupId>
             <artifactId>gson</artifactId>
-            <version>2.6.2</version>
+            <version>2.8.6</version>
```
Thanks for bumping. I guess we can even set this dependency to `provided` as it is in bukkit already

Krymonota replied:
Right, that's something we could do.

Just a side note: Since this project supports multiple versions of Bukkit, it would be better to compile **and relocate** this dependency for compatibility reasons. This would ensure that a new Gson version (which could be shipped with a new version of the Minecraft vanilla server) - or even a modified version of Mojang - wouldn't cause conflicts due to breaking changes. Since Gson is only used to a very small and simple extent here, and it's unlikely that the Gson API used in this project will change in the near future, this risk can be neglected for now.

Krymonota committed:
Specify version (3.8.1) of maven-compiler-plugin

It's never a good idea not to specify the version, because then it
refers to the latest version that could break the build process when a
new version is released.

[WARNING] Some problems were encountered while building the effective
model for net.badlion:blccpsapibungee:jar:1.2.1
[WARNING] 'build.plugins.plugin.version' for
org.apache.maven.plugins:maven-compiler-plugin is missing. @ line 34,
column 21
[WARNING]
[WARNING] Some problems were encountered while building the effective
model for net.badlion:blccpsapibukkit:jar:1.2.1
[WARNING] 'build.plugins.plugin.version' for
org.apache.maven.plugins:maven-compiler-plugin is missing. @ line 34,
column 21
[WARNING]
[WARNING] It is highly recommended to fix these problems because they
threaten the stability of your build.
[WARNING]
[WARNING] For this reason, future Maven versions might no longer support
building such malformed projects.

PSNRigner commented:
This project has been moved to https://github.com/BadlionClient/BadlionClientModAPI and Velocity support has been added.
